myokit/formats/sbml/_importer.py

```diff
@@ -203,89 +291,102 @@ def _parse_parameters(self, model, comp, refs, node):
+                if child:
+                    var.set_rhs(parse_mathml_etree(
+                        child,
+                        lambda x, y: myokit.Name(refs[x]),
+                        lambda x, y: myokit.Number(x)
+                    )
+                    )
             else:
                 raise SBMLError(   # pragma: no cover
                     'Assignment found for unknown parameter: "' + var + '".')
-            node = dom_next(node, 'assignmentRule')
 
         # Create variables with rate rules (states)
-        node = dom_child(parent, 'rateRule')
-        while node:
+        nodes = parent.findall(ns + 'rateRule')
+        for node in nodes:
             var = self._convert_name(
-                str(node.getAttribute('variable')).strip())
+                str(node.get('variable')).strip())
             if var in comp:
                 self.logger().log('Parsing rate rule for <' + var + '>.')
                 var = comp[var]
                 ini = var.rhs()
                 ini = ini.eval() if ini else 0
                 var.promote(ini)
-                var.set_rhs(parse_mathml_dom(
-                    dom_child(node, 'math'), refs, self.logger()))
+                # get child
+                child = node.find(mathml_ns + 'math')
+                # add expression to model
+                if child:
+                    var.set_rhs(parse_mathml_etree(
+                        child,
+                        lambda x, y: myokit.Name(refs[x]),
+                        lambda x, y: myokit.Number(x)
+                    )
+                    )
             else:
                 raise SBMLError(   # pragma: no cover
                     'Derivative found for unknown parameter: <' + var + '>.')
-            node = dom_next(node, 'rateRule')
 
-    def _parse_units(self, model, comp, node):
+    def _parse_units(self, model, comp, node, ns):
         """
         Parses custom unit definitions, creating a look-up table that can be
         used to convert these units to myokit ones.
         """
-        node = dom_child(node, 'unitDefinition')
-        while node:
-            name = node.getAttribute('id')
+        nodes = node.findall(ns + 'unitDefinition')
+        for node in nodes:
+            name = node.get('id')
             self.logger().log('Parsing unit definition for "' + name + '".')
             unit = myokit.units.dimensionless
-            node2 = dom_child(node, 'listOfUnits')
-            node2 = dom_child(node2, 'unit')
-            while node2:
-                kind = str(node2.getAttribute('kind')).strip()
+            units = node.find(ns + 'listOfUnits')
+            units = units.findall(ns + 'unit')
+            for node2 in units:
+                kind = str(node2.get('kind')).strip()
                 u2 = self._convert_unit(kind)
-                if node2.hasAttribute('multiplier'):
-                    m = float(node2.getAttribute('multiplier'))
+                if node2.get('multiplier'):
```

Comment:
This looks like it could be more efficient?
What does node2.get('multiplier') do if multiplier isn't set?

Reply:
node.get() returns None if multiplier does not exist, so it wouldn't do anything.

Reply:
I'm not 100 percent sure how myokit units behave, but would something like this produce the same 
```
kind = str(node2.get('kind')).strip()
u2 = self._convert_unit(kind)
if node2.get('multiplier'):
      u2 *= float(node2.get('multiplier'))
 if node2.get('scale'):
       u2 *= 10 ** float(node2.get('scale'))
 if node2.get('exponent'):
        u2 **= float(node2.get('exponent'))
 unit *= u2
```

And what happens anyway if a unit has several `kind`. I wanted to clarify these questions for the next pull request and just kept your structure and simply replaced minidom by etree everywhere.

Reply:
Fair enough. I'm just not sure if get() is an expensive operation. Probably not?
I was thinking `x = get(); if x: ...` would give us a single get
Up to you :D

Reply:
Actually, are there default options specified for these attributes? `.get()` has a `default` option, so you might be able to do something like
`u2 *= float(node2.get('multiplier', default=1))`
etc.

Reply:
Done